core/controllers/reader.py

```diff
@@ -601,10 +601,12 @@ def post(self, exploration_id):
 
         if user_id and story_id:
             story = story_fetchers.get_story_by_id(story_id)
-            learner_progress_services.mark_story_as_incomplete(
-                user_id, story.id)
-            learner_progress_services.mark_topic_as_partially_learnt(
-                user_id, story.corresponding_topic_id)
+            if story:
+                learner_progress_services.mark_story_as_incomplete(
+                    user_id, story.id)
+            if story.corresponding_topic_id:
```

Comment:
Isn't it dangerous to call this if you don't know whether story exists or not?

Do you have a test for where story_id is invalid? If not please add one.

Reply:
I have changed the condition to mark the topic as partially learnt only when the story exists and also added an exception that will be raised when the story_id is not valid.